### PR TITLE
Fixes an exploit with stacking igniters. Refactors some assembly flag oddities. Limits assembly holders at 12 assemblies. 

### LINKED ISSUE #71222: THE MACHINE THE TERROR MACHINE THE LAG MACHINE
## Round ID:

A with a single click can end entire servers everyone must fear its existence and it must be dealt with

In One I think this should be a feature but thats up to the head admins right now I guess and coders this was definitely intended right ;)

## Testmerges:
image

## Reproduction:
Basically unscrewdrive more than 100 igniters the more the more lag it does when used it could even make the server crash with a single click and attach the igniters together making it all at once do an effect that makes shit lag bad

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Soft revert of #71224 , Fixes #71222

Fixes an exploit involving attachment of multiple igniters to one assembly. 
- Multiple igniters or condensers can no longer be attached to the same assembly holder
- Assembly holders have a limit of 12 assemblies maximum
   - I'm not sure if this is too low or limited, I picked it arbitrarily. Please inform me if it could be upped a smidge. 
   - This lag exploit was born because of limitless assembly holders, which is a little silly even with the exploit aside. All that uncapped holders can bring are exploits or bugs, which I feel confident limited can prevent. What use is there even for having so many?
- Cleans up / refactors some aspects of assemblies and assembly holders.
   - Assemblies had a weird wire type flag that was only ever used by signallers, but also used wrong by signallers. I did some scanning of the code and realized that ... a lot of this was just straight up unused, and not even assigned anywhere. 
   - Now, there is a flag assembly flag var, which everything is read off of. Tested it and still seemed to all work fine.

## Why It's Good For The Game

Lag exploits are bad. 

## Changelog

:cl: Melbert
fix: Fixed an exploit involving igniters attached to themselves. Assembly holders are now limited to 12 assemblies maximum, and you cannot attach multiple igniters to the same assembly.
refactor: Refactored some assembly jank, namely in how they pulse and are pulsed. 
/:cl:

### CODE REVIEW

File: code/modules/assembly/assembly.dm [hunk 1]
```diff
@@ -75,24 +70,33 @@
...
-	if(holder && (wire_type & WIRE_PULSE))
-		holder.process_activation(src, 1, 0)
-	if(holder && (wire_type & WIRE_PULSE_SPECIAL))
-		holder.process_activation(src, 0, 1)
+	// otherwise if we're attached to a holder, process the activation of it with our flags
+	else if(holder)
+		holder.process_activation(
+			device = src,
+			normal = (assembly_flags & ASSEMBLY_WIRE_PULSE),
+			special = (assembly_flags & ASSEMBLY_WIRE_PULSE_SPECIAL),
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/assembly/assembly.dm:98]: remove it then

File: code/modules/assembly/assembly.dm [hunk 2]
```diff
@@ -75,24 +70,33 @@
...
-	if(wire_type & WIRE_RECEIVE)
-		INVOKE_ASYNC(src, .proc/activate, pulser)
-	if(radio && (wire_type & WIRE_RADIO_RECEIVE))
+/**
+ * Pulsed: This device was pulsed by another device
+ *
+ * * wire_flags: What flags to check assembly flags for. Should either be ASSEMBLY_WIRE_RECEIVE or ASSEMBLY_WIRE_RADIO_RECEIVE
+ * * pulser: Who triggered the pulse
+ */
+/obj/item/assembly/proc/pulsed(wire_flags = ASSEMBLY_WIRE_RECEIVE, mob/pulser)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/modules/assembly/assembly.dm:79]: remove it then

[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/assembly/assembly.dm:79]: agree

File: code/modules/assembly/assembly.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mothblocks) [reply code/modules/assembly/assembly.dm:98]: agree

**Pull request merged**